AK/Slugify.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019-2020, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/DeprecatedString.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <cctype>
+
+namespace AK {
+
+inline ErrorOr<String> slugify(String const& _raw, char glue = '-')
```

Comment:
1. This being an inline function in a header will make people sad. My limited understanding is it bloats the binary size, but someone like BertalanD could explain that properly.
2. `_raw` isn't a nice variable name. `raw` is fine, just don't start names with an underscore.

Reply:
Splitting implementation to cpp file makes sense here. I will do it.